services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationService.kt

```diff
@@ -427,51 +427,69 @@ open class ApplicationService(
         }
     }
 
-    private fun sendApplicationToAllu(application: ApplicationEntity): Int {
+    private fun sendToAllu(entity: ApplicationEntity): Int {
+        val application = entity.toApplication()
+        val attachments = entity.attachments.map { it.toAlluAttachment() }
         return if (application.alluid == null) {
-            createApplicationInAllu(application)
+            createApplicationInAllu(application, attachments)
         } else {
             updateApplicationInAllu(application)
         }
     }
 
-    private fun updateApplicationInAllu(application: ApplicationEntity): Int {
-        val alluId =
-            application.alluid ?: throw ApplicationArgumentException("AlluId null in update.")
-        logger.info { "Uploading updated application with alluId $alluId" }
+    private fun updateApplicationInAllu(application: Application): Int =
+        with(application) {
```

Comment:
I don't think every function should start with `with() {`. Overall, I'm not the biggest fan of this pattern. I think it makes it less clear which symbols are local variables and which are fields of the object. All for saving a few characters when referencing the object?

Reply:
Not having to reference the object name on every access is exactly what 'with' is meant for. I'll go through this and see if this can / should be changed.

Reply:
Furthermore, shorter is usually better in Kotlin. But I agree that is up to a certain extent.

Reply:
I think this is a valid use for with():
```
    fun toAlluData(
        hankeTunnus: String,
        applicationData: CableReportApplicationData
    ): AlluCableReportApplicationData =
        with(applicationData) {
```
That method is clearly focused on the application data. It doesn't call any class dependencies or use any external values. And the references to it are always in the same position, so it's clear even without IDE colouring.

Shorter is only better if it's also clearer. Clarity, not terseness, is what defines good code.